gcp_variant_transforms/pipeline_common.py

```diff
@@ -361,5 +370,90 @@ def create_output_table(full_table_id, total_base_pairs, schema_file_path):
   result = os.system(bq_command)
   if result != 0:
     raise ValueError(
-        'Failed to create a bigquery table using "{}" command.'.format(
+        'Failed to create a BigQuery table using "{}" command.'.format(
             bq_command))
+
+
+def delete_table(full_table_id):
+  bq_command = _BQ_DELETE_TABLE_COMMAND.format(FULL_TABLE_ID=full_table_id)
+  return os.system(bq_command)
+
+
+def delete_gcs_files(root_path):
+  gcs_command = _GCS_DELETE_FILES_COMMAND.format(ROOT_PATH=root_path)
+  return os.system(gcs_command)
+
+
+class LoadAvro(object):
+  def __init__(self,
+               avro_root_path,  # type: str
+               table_base_name,  # type: str
+               suffixes,  # type: List[str]
+               total_base_pairs  # type: List[int]
+              ):
+    assert len(suffixes) == len(total_base_pairs)
+
+    self._avro_root_path = avro_root_path
+    self._table_base_name = table_base_name.replace(':', '.')
+    self._suffixes = suffixes
+    self._total_base_pairs = total_base_pairs
+
+    self._num_load_jobs_retries = 0
+    self._suffixes_to_load_jobs = {}  # type: Dict[str, bigquery.job.LoadJob]
+    self._remaining_load_jobs = self._suffixes
+
+    self._client = bigquery.Client()
+
+  def start_loading(self):
+    # We run _MAX_NUM_CONCURRENT_BQ_LOAD_JOBS load jobs in parallel.
+    for _ in range(_MAX_NUM_CONCURRENT_BQ_LOAD_JOBS):
+      self._start_one_load_job(self._remaining_load_jobs.pop())
+
+    self._monitor_load_jobs()
+
+  def _start_one_load_job(self, suffix):
+    job_config = bigquery.LoadJobConfig(
+        source_format=bigquery.SourceFormat.AVRO)
+    uri = self._avro_root_path + suffix + '-*'
+    table_id = sample_info_table_schema_generator.compose_table_name(
+        self._table_base_name, suffix)
+    load_job = self._client.load_table_from_uri(
+        uri, table_id, job_config=job_config)
+    self._suffixes_to_load_jobs.update({suffix: load_job})
+
+  def _cancel_all_running_load_jobs(self):
+    for _, load_job in self._suffixes_to_load_jobs.items():
+      load_job.cancel()
+
+  def _handle_failed_load_job(self, suffix, load_job):
+    if self._num_load_jobs_retries < _BQ_LOAD_JOB_NUM_RETRIES:
+      self._num_load_jobs_retries += 1
+      # Retry the failed job after 5 minutes wait.
+      time.sleep(300)
```

Comment:
why such a drastic wait time? doesn't reducing the count of parallel bq jobs do the trick?